src/OrbitGl/GraphTrack.cpp

```diff
@@ -88,20 +89,60 @@ void GraphTrack::Draw(GlCanvas* canvas, PickingMode picking_mode, float z_offset
     return;
   }
 
+  const Color kBlack(0, 0, 0, 255);
+  const Color kWhite(255, 255, 255, 255);
+  float text_z = GlCanvas::kZValueEvent + z_offset;
+
+  // Add warning threshold text box and line.
+  Batcher* batcher = canvas->GetBatcher();
+  if (warning_threshold_.has_value()) {
+    const Color kThresholdColor(244, 67, 54, 255);
+
+    double normalized_value = (warning_threshold_.value().second - min_) * inv_value_range_;
+    float x = pos_[0];
+    float y = pos_[1] - size_[1] + static_cast<float>(normalized_value) * size_[1];
+    Vec2 from(x, y);
+    Vec2 to(x + size_[0], y);
+
+    std::string text = warning_threshold_.value().first;
+    uint32_t font_size = layout_->CalculateZoomedFontSize();
+    float string_width = canvas->GetTextRenderer().GetStringWidth(text.c_str(), font_size);
+    Vec2 text_box_size(string_width, layout_->GetTextBoxHeight());
+    Vec2 text_box_position(pos_[0] + layout_->GetRightMargin(),
+                           y - layout_->GetTextBoxHeight() / 2.f);
+    canvas->GetTextRenderer().AddText(text.c_str(), text_box_position[0],
+                                      text_box_position[1] + layout_->GetTextOffset(), text_z,
+                                      kThresholdColor, font_size, text_box_size[0]);
+
+    batcher->AddLine(from, from + Vec2(layout_->GetRightMargin() / 2.f, 0), text_z,
+                     kThresholdColor);
+    batcher->AddLine(Vec2(text_box_position[0] + text_box_size[0], y), to, text_z, kThresholdColor);
+  }
+
+  // Add value upper bound text box (e.g., the "Memory Total" text box for the memory tracks).
+  if (value_upper_bound_.has_value()) {
+    std::string text = value_upper_bound_.value().first;
+    uint32_t font_size = layout_->CalculateZoomedFontSize();
+    float string_width = canvas->GetTextRenderer().GetStringWidth(text.c_str(), font_size);
+    Vec2 text_box_size(string_width, layout_->GetTextBoxHeight());
+    Vec2 text_box_position(pos_[0] + size_[0] - text_box_size[0] - layout_->GetRightMargin() -
+                               layout_->GetSliderWidth(),
+                           pos_[1] - layout_->GetTextBoxHeight() / 2.f);
+    canvas->GetTextRenderer().AddText(text.c_str(), text_box_position[0],
+                                      text_box_position[1] + layout_->GetTextOffset(), text_z,
+                                      kWhite, font_size, text_box_size[0]);
+  }
+
   // Draw label
   uint64_t current_mouse_time_ns = time_graph_->GetCurrentMouseTimeNs();
   auto previous_point = GetPreviousValueAndTime(current_mouse_time_ns);
-
```

Comment:
This line was erased on purpose?

Reply:
Ah. It is removed by mistake. I added it back for the readability.